src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -174,7 +253,14 @@ def updatePileup(self, doc, rseList=None, validate=True):
             dbDoc = results[0]
             dbDoc.update(doc)
             try:
-                obj = MSPileupObj(dbDoc, validRSEs=rseList)
+                # when input doc spec is partial pileup we do not have rseList
+                if partialPileupSpec:
+                    rseList = dbDoc['expectedRSEs']
```

Comment:
and I answered above, we do need it since when we create MSPileupObj it requires validation of its rses. In other words MSPileupObj takes given document which already has rses and those must be validated.

Reply:
This is precisely my point. If the document already has those RSE names, then I would assume that the RSEs have **already** been validated. Why should we bother validating it once again?

Reply:
We already discussed it via Zoom and clarified that it is required to pass validRSEs due to validation step in MSPileupObj.